dotnet/src/SemanticKernel.Abstractions/Orchestration/FunctionResult.cs

```diff
@@ -81,6 +81,16 @@ public FunctionResult(string functionName, ContextVariables variables, object? v
     /// <exception cref="InvalidCastException">Thrown when it's not possible to cast result value to <typeparamref name="T"/>.</exception>
     public T? GetValue<T>()
     {
+        if (this.IsCancellationRequested)
+        {
+            throw new InvalidOperationException("Cannot get result value from cancelled function.");
```

Comment:
OperationCanceledException?

Reply:
> OperationCanceledException is for the act of cancellation, if this happens after the cancellation make sense to be an InvalidOperationException

This feels similar to me as async/await.  If you do:
```C#
Task t = SomethingAsync();
...
var result = await t;
```
and SomethingAsync is canceled, the cancellation exception propagates out when you go to get the result.